src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -436,7 +436,7 @@ def load_arguments(self, _):
         c.argument('tier',
                    arg_type=tier_param_type,
                    help='The edition component of the sku. Allowed values include: Basic, Standard, '
-                   'Premium, GeneralPurpose, BusinessCritical.')
+                   'Premium, GeneralPurpose, BusinessCritical, Hyperscale.')
```

Comment:
here the allowed values change, right? There two small questions here:
1. can you try to use get_enum_type() for allowed values?
2. given more supported values here, it will be better to declare it in release note.

Reply:
I am deliberately not using get_enum_type() because there are some values of in `DatabaseEdition` that we don't want to allow for this specific command. I also am not declaring in release note because it's only a documentation change. Hyperscale was already supported.

Reply:
you can customize the allowed values using get_enum_type(["Basic", "Standard", "Premium", "GeneralPurpose", "BusinessCritical", "Hyperscale"]). We will recommend to use get_enum_type() because it will follow the same string format and case insensitive for parameter values.

Reply:
Customizing the list is good, because that will allow us to hide the editions that aren't supported. However we still have the problem of forward compatibility: we are likely to release new editions in the future, so we would like the ability for the user to type in any value that they like so that today's version of CLI has support for future editions.

Reply:
Just to clarify, my understanding is that get_enum_list() will force the value to be validated in client-side. We want to avoid the validation so that "unknown values" are still passed to the server and let the server decide if it's valid.

Reply:
Although I can understand your concern, users should only input values allowed as shown in help message. 
get_enum_type() will check allowed values but it also can help transform enum values to allowed ones, which is very convenient to users.

Reply:
@jaredmoo any thought here?

Reply:
For us, the forward compatibility is important. We don't want to force customers to upgrade their CLI just to access new editions. This is why get_enum_type() doesn't work for us in this scenario.

Reply:
Got it.